### PR TITLE
fix(arcgis): DeckGL rendering integration

### PR DESCRIPTION
**Pull request by jaredscarr**
For #8428.

#### Background
This PR fixes the integration between Arcgis and deckgl. The updates in the Arcgis module fixes the example in `examples/get-started/pure-js/arcgis`. The core arcgis library was updated so that webgl2 was used. The render to screen shader geometry and bindings were updated. Finally, the texture from the `deckInstance` reconstructed so that it could receive data.

https://github.com/visgl/deck.gl/assets/14153643/616ac58d-e7e2-4788-ab07-9d4f19bde283

#### Change List

* examples/get-started/pure-js/arcgis/package.json
  - Updated `@arcgis/core` 
* modules/arcgis/package.json
  - Updated `@arcgis/core`
* modules/arcgis/src/commons.ts
  - Texture creation
  - shader geometry
  - deckFbo added required colorAttachments property
  - bindings and rendering reconstructed
* modules/arcgis/src/deck-layer-view-2d.js
  - Passes the device with the gl context attached

### CODE REVIEW

**Comment by jaredscarr** [comment]
@felixpalmer Context is now webgl2, the device is correct, and the texture binding seems to work. However, the screen is black which I did not expect. Any thoughts on where I should start debugging this part? Thank you for your time and feedback.

**Comment by jaredscarr** [comment]
Made some progress, we've narrowed down the possible problem. We rendered the texture coordinates which confirms that the DeckGL shader is rendering over ArcGis

gradient

We also confirmed our texture bindings are correct, we loaded an image from a url, created a texture from it, passed it to the shader, and it renders as expected

https://github.com/visgl/deck.gl/assets/14153643/8f008d88-11bc-4bd4-bf5f-a0d66c2660ec

When we try to draw with deckFbo in the shader it's all black, so that implies the deckInstance isn't actually writing its output to the fbo. We think the way it's supposed to do that is in this [block](https://github.com/jaredscarr/deck.gl/blob/a79dc2bdb7325f6597cbce4ca09e46657d102d8a/modules/arcgis/src/commons.ts#L83-L98)

```javascript
    // This deck renders into an auxiliary framebuffer.    
     _framebuffer: this.deckFbo,
```

Do you have any tips on further debugging this? is there an example somewhere in the codebase/with the new v9 API of taking the output from a deckinstance and putting it into a framebuffer/texture? Thank you for your time.

**Comment by OmarShehata** [comment]
@felixpalmer Jared has made some progress here, I think we're fairly certain that at least one broken thing is the render to texture part. Are there any examples using luma v9 of how to correctly issue a draw call and render the output into a texture/fbo?

**Comment by felixpalmer** [comment]
@OmarShehata I was going to suggest the `test/apps/mask` along with enabling `debugFBO` in `mask-effect.ts`, however I tried running that just now and it isn't working. So, either you can wait while we figure out how to fix this, or try and use an older beta of luma (as this example was definitely working). I think it's better to wait, it should be fixed within a few days

**Comment by felixpalmer** [comment]
@OmarShehata @jaredscarr https://github.com/visgl/deck.gl/pull/8599

**Comment by jaredscarr** [comment]
@felixpalmer Please take a look as the integration is now working. I've updated the PR description to reflect the changes.

File: modules/arcgis/src/commons.ts [hunk 1]
```diff
@@ -89,18 +128,27 @@ export function render({gl, width, height, viewState}) {
   this.deckInstance.redraw('arcgis');
 
   // We overlay the texture on top of the map using the full-screen quad.
-  const device: Device = this.deckInstance.deck.device;
+  const device: WebGLDevice = this.deckInstance.device;
+
+  const textureToScreenPass = device.beginRenderPass({
+    framebuffer: screenFbo,
+    parameters: {viewport: [0, 0, width, height]},
```
[the remark this answers is not in the document]
Comment (jaredscarr) [reply modules/arcgis/src/commons.ts:135]: Which parameter are you referring to specifically?

File: modules/arcgis/src/commons.ts [hunk 2]
```diff
@@ -3,12 +3,38 @@
...
+    height: 1,
+    sampler: {
+      minFilter: 'linear',
+      magFilter: 'linear',
+      addressModeU: 'clamp-to-edge',
+      addressModeV: 'clamp-to-edge'
+    }
+  });
+
+  this.deckglTexture = deckglTexture;
```
[the remark this answers is not in the document]
Comment (jaredscarr) [reply modules/arcgis/src/commons.ts:24]: done

File: modules/arcgis/src/commons.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jaredscarr) [reply modules/arcgis/src/commons.ts:125]: removed in lower to be consistent with were the duplicate framebuffer was removed

File: modules/arcgis/package.json [hunk 3]
```diff
@@ -40,7 +40,7 @@
     "esri-loader": "^3.3.0"
   },
   "peerDependencies": {
-    "@arcgis/core": "^4.0.0",
+    "@arcgis/core": "^4.28.0",
```
Comment (Pessimistress) [inline modules/arcgis/package.json:43]: Is there anything specifically depending on a feature in this version? Better keep peer dependency restrictions loose for best compatibility.

**Review by Pessimistress — APPROVED** [review]

File: modules/arcgis/package.json (hunk 3 above)
Comment (jaredscarr) [reply modules/arcgis/package.json:43]: At the beginning of this work it appeared that an update would be required in order to use the webgl2 context. However, that was before the context was attached to the device. It seems to work fine if I downgrade at this point.

**Comment by felixpalmer** [comment]
Great work @jaredscarr 😊

**Pull request merged**